views/admin/user/add-user.php

```diff
@@ -94,7 +94,129 @@ function validateEmail($email) {
+
+    $sql = "SELECT * FROM User WHERE user_id = :user_id";
+    $stm = $conn->prepare($sql);
+    $stm->bindValue(":user_id", $_GET["id"], PDO::PARAM_INT);
+
+    $stm->execute();
+    if ($user = $stm->fetch(PDO::FETCH_ASSOC)) {
+        $userId = $user["user_id"];
+        $username = $user["name"];
+        $email = $user["email"];
+        $password = $user["password"];
+        $roomId = $user["room_id"];
+        $extraInfo = $user["extra_info"];
+        $picture = $user["avatar"];
+    } else {
+        $_SESSION["error"] = "This user does not exist in the database.";
+        $PAGE_TITLE="Add user";
+        $pageHeader = "Add user";
+        $editMode = false;
+    }
+}
+
+if (isset($_POST["submit"]) && $_POST["submit"] == "Edit") {
+    if (empty($_POST["username"])) {
+        $_SESSION["error"] = "Username is required.";
+        header("Location: /cafeteria/views/admin/user/add-user.php?id=" . $userId);
+        return;
+    } else {
+        $username = $_POST["username"];
     }
+
+    if (empty($_POST["email"]) || !validateEmail($_POST["email"])) {
+        $_SESSION["error"] = "Invalid email address.";
+        header("Location: /cafeteria/views/admin/user/add-user.php?id=" . $userId);
+        return;
+    } else {
+        $email = $_POST["email"];
+    }
+
+    if (empty($_POST["password"]) || !validatePassword($_POST["password"])) {
+        $_SESSION["error"] = "Password must be not less than 8 characters.";
+        header("Location: /cafeteria/views/admin/user/add-user.php?id=" . $userId);
+        return;
+    } else {
+        $password = md5($_POST["password"]);
+    }
+
+    if (empty($_POST["confirm-password"]) || md5($_POST["confirm-password"]) != $password) {
+        $_SESSION["error"] = "Password and confirm password must match.";
+        header("Location: /cafeteria/views/admin/user/add-user.php?id=" . $userId);
+        return;
+    }
+
+    if (empty($_POST["room-id"]) || $_POST["room-id"] == 0) {
+        $_SESSION["error"] = "Room name is required";
+        header("Location: /cafeteria/views/admin/user/add-user.php");
+        return;
+    } else {
+        $roomId = $_POST["room-id"];
+    }
+
+    $extraInfo = $_POST["extra-info"];
+
+    // Upload user picture.
+    if (isset($_FILES["picture"])) {
+        $newPicture = $_FILES["picture"];
+        $dir = "../../../images/avatars/";
+        $pictureName = $_POST["email"] . "." . strtolower(pathinfo($newPicture["name"], PATHINFO_EXTENSION));
+        $path = $dir . $pictureName;
+
+        if ($newPicture["size"] == 0 || $newPicture["size"] > 5000000) {
+            $_SESSION["error"] = "Picture is too large choose another one and try again.";
+            header("Location: /cafeteria/views/admin/user/add-user.php");
+            return;
+        }
+        if (!empty($picture)) {
+            unlink($dir . $picture);
+        }
+        move_uploaded_file($newPicture["tmp_name"], $path);
+        $picture = $pictureName;
+    }
```

Comment:
well , What About user try to upload pdf file or txt or any extension else ?

Reply:
Ok, I'll work on it